src/server/qgis_mapserver.cpp

```diff
@@ -357,6 +348,23 @@ int main( int argc, char *argv[] )
             }
           }
 
+          // Build URL from env ...
+          QString url { qgetenv( "REQUEST_URI" ) };
+          // ... or from server ip/port and request path
+          if ( url.isEmpty() )
+          {
+            const QString path { firstLinePieces.at( 1 )};
+            // Take Host header if defined
+            if ( headers.contains( QStringLiteral( "Host" ) ) )
+            {
+              url = QStringLiteral( "http://%1%2" ).arg( headers.value( QStringLiteral( "Host" ) ) ).arg( path );
```

Comment:
What about https?

Reply:
Nope, I though about it and since HTTPS is not supported by the 15 LOC web server I wrote it does not make sense here, but if you really want to use this standalone thing behind something that handles the certificates (which I would strongly advise against) your reverse proxy (nginx?) would need to take care take care of setting the `HOST` header.